common/src/main/scala/org/apache/celeborn/common/CelebornConf.scala

```diff
@@ -1116,7 +1116,23 @@ class CelebornConf(loadDefaults: Boolean) extends Cloneable with Logging with Se
   // //////////////////////////////////////////////////////
   //               Authentication                        //
   // //////////////////////////////////////////////////////
-  def authEnabled: Boolean = get(AUTH_ENABLED)
+  def authEnabled: Boolean = {
+    val authEnabled = get(AUTH_ENABLED)
+    val internalPortEnabled = get(INTERNAL_PORT_ENABLED)
+    if (authEnabled && !internalPortEnabled) {
+      throw new IllegalArgumentException(
+        s"${AUTH_ENABLED.key} is true, but ${INTERNAL_PORT_ENABLED.key} is false")
```

Comment:
what are all valid combinations of AUTH_ENABLED and INTERNAL_PORT_ENABLED?
- true, true
- false, false

what if we eliminate INTERNAL_PORT_ENABLED and just respect AUTH_ENABLED?

Reply:
Another valid combination is auth_enabled = false and internal_port_enabled =true.
Having Masters and workers communicate on a separate port is a distinct feature from authentication. In a prior discussion with @waitinfuture, they were considering adding a separate port for internal communication for different reasons. However, it's important to note that this separate  internal port is a prerequisite for authentication.

Reply:
thanks for explanation